web/ui/react-app/src/ExpressionInput.tsx

```diff
@@ -110,6 +133,8 @@ class ExpressionInput extends Component<ExpressionInputProps> {
                 </InputGroupText>
               </InputGroupAddon>
               <Input
+                onInput={this.handleInput}
```

Comment:
Yes, this is not as it should be and as far I can see downshift the reason. I'm gonna investigate it later on

Reply:
Yeah, I had a lot of trouble getting Downshift to behave just the way I wanted to... in case this gets too difficult, it might be worth seeing what happens to https://github.com/prometheus/prometheus/issues/6160. In case @slrtbtfs will really manage to replace the whole expression input field soon, then it might not be worth putting too much effort into it. But that depends on if he will be able to get to it...